### DISCUSSION TITLE
`useMutation` - undocumented `variables` returned by `useMutation`

### DISCUSSION THREAD

**Discussion by mathieu-anderson** [TanStack/query#5082] [Q&A] [answered]
I recently discovered that `useMutation` has `variables` in the list of returned values. Demo sandbox: https://codesandbox.io/s/relaxed-leftpad-nmuhxy?file=/src/App.js

My question is: why is it not documented here? https://tanstack.com/query/v4/docs/react/reference/useMutation

Should it not be used? We have a few use cases where it would come in handy. For example, in a list of buttons which all call the same `mutate` but with a different variable, it would be useful to know exactly which button should show a loading state when `isLoading` is `true`

**Comment by TkDodo** [comment, marked as answer]
You can use it, I'm not sure if I changed the behaviour slightly with v5, because what you are describing will be a first class pattern in v5:

https://tanstack.com/query/v5/docs/react/guides/optimistic-updates#via-the-ui

I just realised that I forgot to update the v5 reference docs to mention variables. If you want, you can contribute that to the `alpha` branch.

**Reply by mathieu-anderson** [reply]
Thank you! Which contribution would be desirable? Adding the `variables` returned value to [the v5 `useMutation` docs](  https://tanstack.com/query/v5/docs/react/reference/useMutation)? Or adding the pattern I described in the "Guides & Concepts" section? Or both? Or neither?

**Reply by TkDodo** [reply]
I think we have a good description in the `guides` already, so I'd appreciate it if you'd get the reference docs in a correct state 🙏

**Reply by mathieu-anderson** [reply]
Opened a PR against the `alpha` branch here: https://github.com/TanStack/query/pull/5084
